spec/monitoring/middleware/conjur_collector_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+require 'rack/test'
+require 'monitoring/prometheus'
+require 'monitoring/middleware/conjur_collector'
+require 'monitoring/middleware/prometheus_exporter'
+
+describe Monitoring::Middleware::ConjurCollector do
+  include Rack::Test::Methods
+
+  before do
+    Monitoring::Prometheus.setup(registry: Prometheus::Client::Registry.new)
+  end
+
+  let(:registry) { Monitoring::Prometheus.registry }
+  let(:metric) { :collector_test_metric }
+  let(:labels) { { path: '/foo', code: 200 } }
+  let(:exporter_options) { { registry: registry, path: '/metrics' } }
+
+  let(:app) do
+    app = ->(_) { [200, { 'Content-Type' => 'text/html' }, ['OK']] }
+
+    # Wrap the mock app with the Collector and Exporter middleware
+    app_with_collector = described_class.new(app)
+    Monitoring::Middleware::PrometheusExporter.new(app_with_collector, **exporter_options)
+  end
+
+  it 'returns the app response' do
+    get '/foo'
+    expect(last_response).to be_ok
+    expect(last_response.body).to eql('OK')
+  end
+
+  it 'metrics from traced requests update the global registry' do
```

Comment:
This test looks a lot like an integration test.  It's testing that the `record` method correctly publishes an event (and the event is picked up in the registry).  Although this is fine (and certainly has its place in testing), we're skipping tests on all the methods in `Monitoring::Middleware::ConjurCollector`.  

Can you please include some more granular tests to verify happy and sad paths of the methods in `ConjurCollector`?